Mathlib/Tactic/Linter/Lint.lean

```diff
@@ -211,6 +211,43 @@ initialize addLinter cdotLinter
 
 end CDotLinter
 
+/-!
+# The `dollarSyntax` linter
+
+The `dollarSyntax` linter flags uses of `<|` that are achieved by typing `$`.
+-/
+
+/-- The `dollarSyntax` linter flags uses of `<|` that are achieved by typing `$`. -/
+register_option linter.dollarSyntax : Bool := {
+  defValue := true
+  descr := "enable the `dollarSyntax` linter"
+}
+
+namespace DollarSyntaxLinter
+
+/-- `findDollarSyntax stx` extracts from `stx` the syntax nodes of `kind` `$`. -/
+partial
+def findDollarSyntax : Syntax → Array Syntax
+  | stx@(.node _ kind args) =>
+    let dargs := (args.map findDollarSyntax).flatten
+    match kind with
+      | ``«term_$__» => dargs.push stx
+      | _ => dargs
+  |_ => #[]
+
+@[inherit_doc linter.dollarSyntax]
+def dollarSyntaxLinter : Linter where run := withSetOptionIn fun stx ↦ do
+    unless Linter.getLinterValue linter.dollarSyntax (← getOptions) do
+      return
+    if (← MonadState.get).messages.hasErrors then
+      return
+    for s in findDollarSyntax stx do
+      Linter.logLint linter.dollarSyntax s m!"Please use '<|' and not '$' for the pipe operator."
```

Comment:
I like to think carefully about linter wording: what do you think about the following change?
```suggestion
      Linter.logLint linter.dollarSyntax s m!"Please use '<|' instead of '$' for the pipe operator."